src/index/handler/git.rs

```diff
@@ -0,0 +1,349 @@
+use crate::index::handler::{IndexPublishError, IndexPublisher};
+use async_trait::async_trait;
+use forc_pkg::source::reg::{
+    file_location::{location_from_root, Namespace},
+    index_file::{IndexFile, PackageEntry},
+};
+use git2::{Cred, FetchOptions, Oid, PushOptions, RemoteCallbacks, Repository, Signature};
+use std::env;
+use std::fs;
+use std::path::Path;
+use tempfile::TempDir;
+use tokio::task;
+
+/// Index publishing backend for GitHub.
+pub struct GithubIndexPublisher {
+    repo_owner: String,
+    repo_name: String,
+    chunk_size: usize,
+    namespace: Namespace,
+}
+
+impl GithubIndexPublisher {
```

Comment:
It would be great to see unit tests for the functions in this module

Reply:
Most of these functions are relying to be able to fetch something which needs internet connection and some env setup to authenticate so it is very hard to test the majority of the stuff. (Possibly not possible as we do cannot add an `SSH_KEY` that has write access to a github repo during tests.)

I am thinking of ways to to tests this but it is not going to be straightforward unfortunatly. If you have suggestions happy to implement them